### ISSUE TITLE
BruteForceSampler crashes with named hyperparameters out of alphabetical order

### ISSUE THREAD

**Issue by dkalsan** [optuna/optuna#4771]
### Expected behavior

The `BruteForceSampler` should run regardless of the alphabetical order of `name`s on repeated invocations of `suggest_float(...)`.

### Environment

- Optuna version: 3.2.0
- Python version: 3.10.6
- OS: Linux-5.15.0-71-generic-x86_64-with-glibc2.35
- (Optional) Other libraries and their versions: /

### Error messages, stack traces, or logs

```shell
Traceback (most recent call last):
  File "/home/user/Desktop/minimal-example.py", line 14, in <module>
    study.optimize(objective, n_trials=100)
  File "/home/user/.virtualenvs/tmp/lib/python3.10/site-packages/optuna/study/study.py", line 443, in optimize
    _optimize(
  File "/home/user/.virtualenvs/tmp/lib/python3.10/site-packages/optuna/study/_optimize.py", line 66, in _optimize
    _optimize_sequential(
  File "/home/user/.virtualenvs/tmp/lib/python3.10/site-packages/optuna/study/_optimize.py", line 163, in _optimize_sequential
    frozen_trial = _run_trial(study, func, catch)
  File "/home/user/.virtualenvs/tmp/lib/python3.10/site-packages/optuna/study/_optimize.py", line 212, in _run_trial
    frozen_trial = _tell_with_warning(
  File "/home/user/.virtualenvs/tmp/lib/python3.10/site-packages/optuna/study/_tell.py", line 174, in _tell_with_warning
    study.sampler.after_trial(study, frozen_trial, state, values)
  File "/home/user/.virtualenvs/tmp/lib/python3.10/site-packages/optuna/samplers/_brute_force.py", line 213, in after_trial
    tree = self._build_tree(
  File "/home/user/.virtualenvs/tmp/lib/python3.10/site-packages/optuna/samplers/_brute_force.py", line 167, in _build_tree
    leaf.set_leaf()
  File "/home/user/.virtualenvs/tmp/lib/python3.10/site-packages/optuna/samplers/_brute_force.py", line 53, in set_leaf
    self.expand(None, [])
  File "/home/user/.virtualenvs/tmp/lib/python3.10/site-packages/optuna/samplers/_brute_force.py", line 46, in expand
    raise ValueError(f"param_name mismatch: {self.param_name} != {param_name}")
```

### Steps to reproduce

```
import optuna
from optuna.samplers import BruteForceSampler


def objective(trial):
    b = trial.suggest_float(name="b", low=0, high=1, step=0.1)  # Line B
    a = trial.suggest_float(name="a", low=0, high=1, step=0.1)  # Line A
    return a * b

study = optuna.create_study(study_name="minimal-example",
                            storage="sqlite:///minimal-example.db",
                            sampler=BruteForceSampler())

study.optimize(objective, n_trials=100)
```

### Additional context (optional)

* The bug is present only when the study is created with the `storage` parameter.
* If lines A and B are switched, then the code runs without errors. Thus the assumption is that the bug has to do with alphabetical order of `name`s of suggested values.

**Comment by not522** [comment]
Thank you for your bug report!
I found the cause of this error and started the discussion with other maintainers.

**Comment by contramundum53** [comment]
As a workaround in optuna v3.2.0, you can use `JournalStorage` instead of SQLite.

**Comment by dkalsan** [comment]
Perfect, thanks a lot!

**Comment by not522** [comment]
This issue has been resolved by #4775, and it will be included the next minor release. Thank you.
